common/src/main/scala/explore/components/Tile.scala

```diff
@@ -29,7 +33,8 @@ final case class Tile(
   canMinimize:       Boolean = false,
   canMaximize:       Boolean = false,
   state:             TileSizeState = TileSizeState.Normal,
-  sizeStateCallback: TileSizeState ~=> Callback = Reusable.always(_ => Callback.empty)
+  sizeStateCallback: TileSizeState ~=> Callback = Reusable.always(_ => Callback.empty),
+  key:               js.UndefOr[Key] = js.undefined
```

Comment:
Tiles can now keyed. This is used to force a remount when switching an observation but one of the tiles references the same object as the previous observation.